cozy/codegen/cxx.py

```diff
@@ -274,6 +274,13 @@ def construct_concrete(self, t : Type, e : Exp, out : Exp):
             return SEscape("{indent}{lhs} = {rhs};\n", ["lhs", "rhs"], [out, e])
         elif is_scalar(t):
             return SEscape("{indent}{lhs} = {rhs};\n", ["lhs", "rhs"], [out, e])
+        elif isinstance(t, TTuple):
+            x = self.fv(t, "x")
+            return SSeq(
+                SEscape("{indent}{rv} = {rhs};\n", ["rv", "rhs"], [x, e]),
+                SEscape("{indent}{lhs} = {rv};\n", ["lhs", "rv"],
+                        [out, ETuple([ETupleGet(x, ZERO.with_type(INT)), ETupleGet(x, ONE.with_type(INT))]).with_type(t)]),
```

Comment:
A few notes:
 - Tuples aren't necessarily pairs. You need to inspect `t` to find out how many elements there are.
 - `ETupleGet` takes two parameters: a tuple-type expression and an integer. Not an expression; an actual integer like `0` or `1`.
 - `ZERO.with_type(INT)` is dangerous because you are mutating the global `ZERO` object. It happens to be safe here since `ZERO` already has type `Int`. This problem will go away when you replace this with `0` instead.